cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -1119,7 +1117,7 @@ private void authorizePipelineRuntimeImpersonation(Map<String, String> userArgs)
     if ((userArgs.containsKey(SystemArguments.RUNTIME_PRINCIPAL_NAME)) &&
             (userArgs.containsKey(SystemArguments.RUNTIME_KEYTAB_PATH))) {
       String principal = userArgs.get(SystemArguments.RUNTIME_PRINCIPAL_NAME);
-      LOG.debug("Checking authorisation for user: %s, using runtime config principal: %s",
+      LOG.debug("Checking authorisation for user: {}, using runtime config principal: {}",
```

Comment:
:)

As a side note, I wonder if there's a way to configure checkstyle to fail on these? Have seen this -- and {} used in String.format() -- several times before.

Reply:
Will investigate.